### PR TITLE
Add doc content for configuring mtls between components

### PR DESCRIPTION
**Pull request by Mirage20**
## Purpose
Add doc content for configuring mtls between components

Related: https://github.com/wso2-enterprise/choreo/issues/20167

### CODE REVIEW

File: en/docs/develop-components/configuring-mutual-tls-between-components.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Configuring mutual TLS between Components
```
Comment (shaniR) [inline en/docs/develop-components/configuring-mutual-tls-between-components.md:1]: Shall  we replace this content with  the following?: 

# Configuring mutual TLS between Components

The Mutual Transport Layer Security (mTLS) is a Transport Layer Security (TLS) protocol that ensures privacy, integrity, and authentication of the data transmitted between two endpoints. In mTLS, both the client and the server authenticate each other using digital certificates, establishing trust and confirming their identities. Upon successful authentication, mTLS encrypts the data exchanged between the client and the server, preventing unauthorized access.

## Configuring mTLS between Components

In Choreo, you can use mTLS to establish secure connections between components within a Choreo Project.

### Certificate Generation

- **Root Certificate:** The root certificate is trusted by both the client and the server. It is used to verify the authenticity of other certificates presented during the mTLS handshake process and issue certificates for clients and servers. For a specific project, you can generate a single root certificate using a tool like OpenSSL.

- **Client Certificate:** The client certificate contains the client's identity, which is used to authenticate the client. You can use the Common Name (CN) in the certificate to provide an identifier for the client. The generated certificate should be signed by the previously generated root certificate.

- **Server Certificate:** Clients can use the server certificate to verify the trustworthiness of the server and establish a secure and authenticated connection. Like client certificates, the server certificate must be signed by the previously generated root certificate. When generating the server certificate, you must specify the server's hostname for the Subject Alternative Name (SAN). You can obtain the hostname for the specific version of a service component from any project endpoint on the **Overview** page.

For example, if your project endpoint is `http://my-service-3781140846:7080/todos`, the hostname will be `my-service-3781140846`.

### Reading mTLS Certificates from your component

The method of reading mTLS certificates from a component may vary depending on its implementation. Typically, a component can read the certificate data from either the file system or via an environment variable. For detailed instructions on adding environment variables and file mounts to your application, refer to [Manage Configurations and Secrets](../devops-and-ci-cd/manage-configurations-and-secrets.md). 

!!! tip
    When providing the **private key**, ensure that you **save it as a Secret**.

## Sample for mTLS Communication

You can find step-by-step instructions on deploying two components in Choreo that utilize mTLS communication in the sample [service-to-service-mtls](https://github.com/wso2/choreo-sample-apps/tree/main/go/service-to-service-mtls).

File: en/mkdocs.yml [hunk 2]
```diff
@@ -104,6 +104,7 @@ nav:
           - Develop a Scheduled Integration: develop-components/develop-integrations/develop-a-scheduled-integration.md
       - Develop a Web Application: develop-components/develop-a-web-application.md       
       - Develop Components With Git: develop-components/develop-components-with-git.md  
+      - Configuring mutual TLS between Components: develop-components/configuring-mutual-tls-between-components.md
```
Comment (shaniR) [inline en/mkdocs.yml:107]: Configure Mutual TLS Between Components

Comment (shaniR) [reply en/mkdocs.yml:107]: Rename the file name to - configure-mutual-tls-between-components.md

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/mkdocs.yml (hunk 2 above)
Comment (Mirage20) [reply en/mkdocs.yml]: Updated with [cee6dc5](https://github.com/wso2/docs-choreo-dev/pull/942/commits/cee6dc53c13522052a972af5b1a9d1d36a359d04)

**Review by shaniR — APPROVED** [review]

**Pull request merged**